plotille/_figure.py

```diff
@@ -324,24 +363,39 @@ def height_vals(self):
         return self.Y
 
     def write(self, canvas, with_colors, in_fmt):
+
+        # plot all points with optional text first, then lines
+        # separating out points from lines allows single-point scatters to successfuly plot
+
+        # make point iterators
+        color = self.lc if with_colors else None
+
+        points = zip(map(in_fmt.convert, self.X), map(in_fmt.convert, self.Y))
+
+        for index, (x, y) in enumerate(points):
+
+            if(isinstance(self.text, list) and index <= len(self.text)):
+                text_for_point = self.text[index]
+            else:
+                text_for_point = self.text
+
+            canvas.point(x, y, color=color, overlay=self.overlay, marker=self.marker, text=text_for_point)
```

Comment:
The `text` of `Plot` is used for every point? This should be the `label` for the legend, right?